pkg/controller/wip/controller_test.go

```diff
@@ -390,6 +390,105 @@ func TestReconcileInstanceNonExistentServiceClass(t *testing.T) {
 	}
 }
 
+func TestReconcileInstanceNonExistentBroker(t *testing.T) {
+	_, fakeCatalogClient, _, _, _, testController, sharedInformers := newTestController(t)
+
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{Name: "test-instance"},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "test-serviceclass",
+			PlanName:         "default",
+			OSBGUID:          instanceGUID,
+		},
+	}
+
+	testController.reconcileInstance(instance)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v. Actions: +%v", e, a, actions)
+	}
+
+	// There should only be one action that says it failed because no such broker exists.
+	updateAction := actions[0].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[0]; expected %v, got %v", e, a)
+	}
+	updateActionObject := updateAction.GetObject().(*v1alpha1.Instance)
+	if e, a := "test-instance", updateActionObject.Name; e != a {
+		t.Fatalf("Unexpected name of instance created: expected %v, got %v", e, a)
+	}
+	if e, a := 1, len(updateActionObject.Status.Conditions); e != a {
+		t.Fatalf("Unexpected number of conditions: expected %v, got %v", e, a)
+	}
+	if e, a := "ReferencesNonexistentBroker", updateActionObject.Status.Conditions[0].Reason; e != a {
+		t.Fatalf("Unexpected condition reason: expected %v, got %v", e, a)
+	}
+}
+
+func TestReconcileInstanceWithAuthError(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, _, _, _, testController, sharedInformers := newTestController(t)
+
+	broker := getTestBroker()
+	broker.Spec.AuthSecret = &v1.ObjectReference{
+		Namespace: "does_not_exist",
+		Name:      "auth-name",
+	}
+	sharedInformers.Brokers().Informer().GetStore().Add(broker)
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{Name: "test-instance"},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "test-serviceclass",
```

Comment:
if I'm following the flow here, can we change "test-serviceclass" to broker.ObjectMetadata.Name (or whatever the field is called), so that we're not hardcoding names?